active/0000-blueprint-remove-old-files.md

```diff
@@ -0,0 +1,48 @@
+- Start Date: 2016-12-17
+- RFC PR: (leave this empty)
+- Ember CLI Issue: (leave this empty)
+
+# Summary
+
+Give blueprint generators the ability to clean up old files.
+
+# Motivation
+
+We want to eliminate the noise of having old files laying after updating
+ember-cli using `ember init`.
+
+# Detailed design
+
+We'd like a declarative API for blueprints to delete files instead of only
+create. It would be essentially syntactic sugar for removing the file yourself
+in an `afterInstall` hook. It would be an array on the blueprint's `index.js`.
```

Comment:
How do the deletion + addition rules compose?

For example:

1. default blueprint: adds `.jshintrc`
2. eslint blueprint:  removes `.jshintrc`


## Scenarios:

### user runs the eslint blueprint for the first time, output would be:

```sh
file remove: .jshintrc
```


### user then later runs `ember init`

One approach would result in:

```sh
file add: .jshintrc
--- eslintrc addon
file remove: .jshintrc
```

One approach would result in no output

```sh
```

One may output exactly what happened:

```sh
file ensured remove: .jshintrc
```